code-challenge/Models/Product.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+struct Product<Result: Decodable>: Decodable {
+    let currentPage: Int
+    let pages: Int
+    let count: Int
+    let results: [Result]
+
+    enum Product: String, CodingKey {
+        case currentPage = "page"
+        case pages = "total_pages"
+        case count = "total_results"
+        case results
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
same here, isn't the auto-generated initializer working for you in this case? why?